packages/manager/modules/pci/src/components/project/stepper-defaults-selection/stepper-defaults-selection.utils.js

```diff
@@ -0,0 +1,62 @@
+import isFunction from 'lodash/isFunction';
+import get from 'lodash/get';
+import set from 'lodash/set';
+
+const getDefaultValue = (field, thisObj) => {
+  let defaultValue;
+  if (thisObj.defaults[field.name] !== undefined) {
+    defaultValue =
+      isFunction(field.getDefault) &&
+      field.getDefault(
+        thisObj.defaults[field.name],
+        thisObj[field.availableOptions],
+      );
+  }
+  return defaultValue;
+};
+
+const onChangeAfterDefaultValueSet = (methodToCall, params, thisObj) => {
+  if (thisObj[methodToCall] && isFunction(thisObj[methodToCall])) {
+    return thisObj[methodToCall].call(
+      thisObj,
+      ...params.map((prop) => get(thisObj, prop)),
+    );
+  }
+  return null;
+};
+
+export const setDefaultSelections = (
+  configObj,
+  DEFAULTS_MODEL,
+  currentStep,
+) => {
+  return new Promise((resolve) => {
```

Comment:
This looks ... complicated, could we simplify ? Also we override params value maybe we could return a new object

Reply:
Hello @marie-j, 
We cannot return a new object here since the controller instance ("this" is the param passed from the controller) is passed to this utility.